src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_message_is_moved_to_error_queue_using_dtc.cs

```diff
@@ -0,0 +1,102 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_message_is_moved_to_error_queue_using_dtc : NServiceBusAcceptanceTest
```

Comment:
> We do want to test that no side-effects are commited when doing SLR/Errors

I can see why you want to test that here based on the related bug. But I'm still not entirely convinced this is the right way to do it. In that case we have to test all the side effects for FLR, SLR and the error queue for all different transaction modes.

e.g. testing for no outgoing messages in FLR/SLR when transaction mode < MultiQueueTx is wrong. It's not always the transaction which does rollback the messages, it's the batching feature and in case something goes wrong after batching, we have outgoing messages when not using MultiQueue Transactions, that's why I hesitate to mix up these different behaviors because it makes things really confusing.

Retries don't do anything to transactions. Retries tell the transport to not consume the message. The transaction is the mechanism implementing this. Retries don't ensure other databases or outgoing messages are enlisted in transactions if possible. Retries ensure the message will be processed again on a processing failure. I see that the existing tests obviously weren't good enough to account for the wrong behavior. But if we had tested whether retries tell the transport API to not consume the message, we would have found that bug in the buggy implementation too, without testing for outgoing messages or transactions.

It's kinda a discussion between integration vs. component tests. I see that the tests make sense from an user perspective (and therefore seems a valid acceptance test). But they are also significantly harder to understand to the full level or may be misleading (see batched dispatch using single queue transactions). Personally I'leaning toward component here but that's highly subjective.